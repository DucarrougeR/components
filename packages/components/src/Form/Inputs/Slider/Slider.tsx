import styled from 'styled-components'
import { reset, space, SpaceProps } from '@looker/design-tokens'
import { InputProps } from '../InputProps'

export interface SliderProps extends SpaceProps, Omit<InputProps, 'type'> {}

export const Slider = styled.input.attrs({ type: 'range' })<SliderProps>`
  ${reset}
  ${space}
`
