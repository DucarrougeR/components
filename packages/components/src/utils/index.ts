export * from './GlobalStyle'
export * from './useToggle'
