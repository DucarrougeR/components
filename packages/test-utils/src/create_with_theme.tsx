import { theme } from '@looker/design-tokens'
import { render } from '@testing-library/react'
import { mount, shallow } from 'enzyme'
import 'jest-styled-components'
import * as React from 'react'
import { create } from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'

export const withThemeProvider = (Component: React.ReactElement<any>) => (
  <ThemeProvider theme={theme}>{Component}</ThemeProvider>
)

export const createWithTheme = (Component: React.ReactElement<any>) =>
  create(withThemeProvider(Component))

export const mountWithTheme = (Component: React.ReactElement<any>) =>
  mount(withThemeProvider(Component))

export const shallowWithTheme = (Component: React.ReactElement<any>) =>
  shallow(withThemeProvider(Component))

export const renderWithTheme = (Component: React.ReactElement<any>) =>
  render(withThemeProvider(Component))
