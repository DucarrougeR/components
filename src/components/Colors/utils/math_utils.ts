/**
 * Utility to convert a numerical value to a percent given the value and a max value.
 */
export const toPerc = (value: number, maxValue: number) =>
  Math.round((value / maxValue) * 100)
