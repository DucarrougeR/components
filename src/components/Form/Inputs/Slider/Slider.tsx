import omit from 'lodash/omit'
import React, { FunctionComponent, Ref } from 'react'
import styled, { StyledComponent } from 'styled-components'
import { Box, BoxProps } from '../../../Box'
import { InputProps } from '../InputProps'

export interface SliderProps extends InputProps, BoxProps<HTMLInputElement> {
  /**
   * Minimum value for Slider.
   */
  min?: number
  /**
   * Maximum value for Slider.
   */
  max?: number
  /**
   * Granularity between values.
   */
  step?: number
  /**
   * Initial value.
   */
  value?: number
}

type ComponentType = FunctionComponent<SliderProps>
type StyledComponentType = StyledComponent<ComponentType, SliderProps>

const InternalSlider: React.FC<SliderProps> = props => {
  return <Box is="input" type="range" {...omit(props, ['validationType'])} />
}

const SliderFactory = React.forwardRef<StyledComponentType, SliderProps>(
  (props: SliderProps, ref: Ref<StyledComponentType>) => (
    <InternalSlider ref={ref} {...props} />
  )
)

/** @component */
export const Slider = styled<ComponentType>(SliderFactory)``
