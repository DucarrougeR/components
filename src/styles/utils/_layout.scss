// Returns a value from the $spacing list
@function spacing($key) {
  @return map-lookup($spacing, $key, '$spacing');
}
