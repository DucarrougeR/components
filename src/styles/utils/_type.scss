// Returns a value from the $test-size list
@function font-size($key) {
  @return map-lookup($text-size, $key, '$text-size');
 }

 // Returns a value from the $line-height list
 @function line-height($key) {
   @return map-lookup($line-height, $key, '$line-height');
 }

 @mixin fontSize($size) {
  font-size: $size;
  font-size: pxToRem($size);
}
