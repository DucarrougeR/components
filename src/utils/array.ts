/**
 * Returns an array of length `size` filled with zeros.
 * Useful for looping constructs where a length is known, but no array is provided.
 */

export const sizedArray = (size: number) => Array(size).fill(0)
