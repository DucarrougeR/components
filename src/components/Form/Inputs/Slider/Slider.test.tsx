import 'jest-styled-components'
import * as React from 'react'
import { mountWithTheme } from '../../../../../test/utils/create_with_theme'
import { assertSnapshot } from '../../../../../test/utils/snapshot'
import { Slider } from './Slider'

test('Slider default', () => {
  assertSnapshot(<Slider />)
})

test('Slider with min and max', () => {
  assertSnapshot(<Slider min={1} max={11} step={0.01} value={11} />)
})

test('Slider with name and id', () => {
  assertSnapshot(<Slider name="Slip" id="Slide" />)
})

test('Slider should accept disabled', () => {
  assertSnapshot(<Slider disabled />)
})

test('Slider with aria-describedby', () => {
  assertSnapshot(<Slider aria-describedby="some-id" />)
})

test('Should trigger onChange handler', () => {
  let counter = 0
  const handleChange = () => counter++
  const wrapper = mountWithTheme(<Slider onChange={handleChange} />)
  wrapper.find('input').simulate('change', { target: { value: '' } })
  expect(counter).toEqual(1)
})
