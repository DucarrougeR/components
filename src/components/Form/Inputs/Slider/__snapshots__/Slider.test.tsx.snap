// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider default 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
}

<input
  className="c0"
  type="range"
/>
`;

exports[`Slider should accept disabled 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
}

<input
  className="c0"
  disabled={true}
  type="range"
/>
`;

exports[`Slider with aria-describedby 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
}

<input
  aria-describedby="some-id"
  className="c0"
  type="range"
/>
`;

exports[`Slider with min and max 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
}

<input
  className="c0"
  max={11}
  min={1}
  step={0.01}
  type="range"
  value={11}
/>
`;

exports[`Slider with name and id 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
}

<input
  className="c0"
  id="Slide"
  name="Slip"
  type="range"
/>
`;
